modules/fedimint-ln-client/src/lib.rs

```diff
@@ -170,22 +169,13 @@ pub enum InternalPayState {
 #[serde(rename_all = "snake_case")]
 pub enum LnPayState {
     Created,
-    Canceled,
-    Funded,
-    WaitingForRefund {
-        block_height: u32,
-        gateway_error: GatewayPayError,
-    },
+    FundingRejected,
+    Funded { block_height: u32 },
+    WaitingForRefund { error_reason: String },
     AwaitingChange,
-    Success {
-        preimage: String,
-    },
-    Refunded {
-        gateway_error: GatewayPayError,
-    },
-    UnexpectedError {
-        error_message: String,
-    },
+    Success { preimage: String },
+    Refunded { error_reason: String },
+    UnexpectedError { error_message: String },
```

Comment:
We had this discussion before, and you are right that we never save this state in the DB consensus encoded, but we do save *final outcomes* in the operation log as JSON.

So I think we need to introduce an alias for `FundingRejected` since it might be saved as `canceled` in the OpLog and give the `Refunded` variant a default error message since the old one has the `GatewayPayError`, which we no longer parse. Eventually we'll want to add migrations for this too.

Reply:
Ah I see. Should I change `FundingRejected` back to `Canceled`? That change is mostly just style, I thought it made more sense, but if it's going to cause issues it probably makes more sense to just keep it as `Canceled`.

How do we add migrations for these states? Would that involve reading the OpLog and re-writing the JSON with it encoded as the new states?

Reply:
> How do we add migrations for these states? Would that involve reading the OpLog and re-writing the JSON with it encoded as the new states?

Yes, I fear so :( but in this case we can probably work around it by having a default error message (and JSON already ignores superfluous fields).

Reply:
Hmm yeah this is more change than I think I want to do with this PR. I'll see if I can change this back so that it is backwards compatible.

Reply:
I changed `FundingRejected` back to `Cancelled` and `Refunded` to contain `GatewayPayError` again. Now only `Funded` and `WaitingForRefund` have new fields.

For my understanding, what is the impact of breaking the JSON in the operation log? Will that impact downstream clients that parse it?